c_api/impl/io_c.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// -*- c++ -*-
+
+#ifndef FAISS_IO_C_H
+#define FAISS_IO_C_H
+
+#include <faiss/impl/io.h>
```

Comment:
C header files must not include C++ header files, otherwise it will break when using a C compiler.

This is probably only needed in `io_c.cpp` anyway.

Reply:
Sorry, didn't realize this -- fixed now